relay-spans/src/span.rs

```diff
@@ -241,140 +152,26 @@ impl From<OtelSpan> for EventSpan {
-    #[default]
-    Internal = 1,
-    /// Indicates that the span covers server-side handling of an RPC or other
-    /// remote network request.
-    Server = 2,
-    /// Indicates that the span describes a request to some remote service.
-    Client = 3,
-    /// Indicates that the span describes a producer sending a message to a broker.
-    /// Unlike CLIENT and SERVER, there is often no direct critical path latency relationship
-    /// between producer and consumer spans. A PRODUCER span ends when the message was accepted
-    /// by the broker while the logical processing of the message might span a much longer time.
-    Producer = 4,
-    /// Indicates that the span describes consumer receiving a message from a broker.
-    /// Like the PRODUCER kind, there is often no direct critical path latency relationship
-    /// between producer and consumer spans.
-    Consumer = 5,
+trait OtelValueExt {
+    fn to_i64(&self) -> Option<i64>;
+    fn to_string(&self) -> Option<String>;
 }
 
-#[derive(Clone, Debug, Deserialize, Default, Serialize)]
-#[serde(rename_all = "camelCase")]
-pub struct Status {
-    /// A developer-facing human readable error message.
-    #[serde(default)]
-    pub message: String,
-    /// The status code.
-    #[serde(default)]
-    pub code: StatusCode,
-}
-
-#[derive(Clone, Copy, Debug, Deserialize_repr, Default, PartialEq, Serialize_repr)]
-#[repr(u8)]
-pub enum StatusCode {
-    /// The default status.
-    #[default]
-    Unset = 0,
-    /// The Span has been validated by an Application developer or Operator to
-    /// have completed successfully.
-    Ok = 1,
-    /// The Span contains an error.
-    Error = 2,
-}
-
-/// AnyValue is used to represent any type of attribute value. AnyValue may contain a
-/// primitive value such as a string or integer or it may contain an arbitrary nested
-/// object containing arrays, key-value lists and primitives.
-/// The value is one of the listed fields. It is valid for all values to be unspecified
-/// in which case this AnyValue is considered to be "empty".
-#[derive(Clone, Debug, Deserialize, Serialize)]
-pub enum AnyValue {
-    #[serde(rename = "arrayValue")]
-    Array(ArrayValue),
-    #[serde(rename = "boolValue")]
-    Bool(bool),
-    #[serde(rename = "bytesValue")]
-    Bytes(Vec<u8>),
-    #[serde(rename = "doubleValue")]
-    Double(f64),
-    #[serde(rename = "intValue")]
-    Int(i64),
-    #[serde(rename = "kvlistValue")]
-    Kvlist(KeyValueList),
-    #[serde(rename = "stringValue")]
-    String(String),
-}
-
-impl AnyValue {
-    pub fn to_i64(&self) -> Option<i64> {
+impl OtelValueExt for OtelValue {
+    fn to_i64(&self) -> Option<i64> {
         match self {
-            AnyValue::Int(v) => Some(*v),
+            OtelValue::IntValue(v) => Some(*v),
             _ => None,
         }
     }
 
-    pub fn to_string(&self) -> Option<String> {
+    fn to_string(&self) -> Option<String> {
```

Comment:
I would prefer if this was something like `into_string` and takes `self`, that means in the `AnyValue::String` case we don't have to clone (and the caller can still clone if he has to, explicitly).